src/java/com/zimbra/oauth/utilities/OAuth2Utilities.java

```diff
@@ -160,4 +169,67 @@ public static Attachment createAttachmentFromResponse(GetMethod method, String k
         }
         return null;
     }
+
+    /**
+     * Executes an Http Request and returns the response body.
+     *
+     * @param request Request to execute
+     * @return Response body as a string
+     * @throws ServiceException If there are issues with the connection
+     * @throws IOException If there are non connection related issues
+     */
+    public static String executeRequest(HttpMethod request)
+        throws ServiceException, IOException {
+        final HttpClient client = OAuth2Utilities.getHttpClient();
+        return executeRequest(client, request);
+    }
+
+    /**
+     * Executes an Http Request with a given client and returns the response body.
+     *
+     * @param client The client to execute with
+     * @param request Request to execute
+     * @return Response body as a string
+     * @throws ServiceException If there are issues with the connection
+     * @throws IOException If there are non connection related issues
+     */
+    public static String executeRequest(HttpClient client, HttpMethod request)
+        throws ServiceException, IOException {
+        String responseBody = null;
+        try {
+            HttpClientUtil.executeMethod(client, request);
+            responseBody = request.getResponseBodyAsString();
+        } catch (final UnknownHostException e) {
+            ZimbraLog.extensions.errorQuietly(
+                "The configured destination address is unknown: " + request.getURI(), e);
+            throw ServiceException
+                .RESOURCE_UNREACHABLE("The configured destination address is unknown.", e);
+        } catch (final SocketTimeoutException e) {
+            ZimbraLog.extensions
+                .warn("The destination server took too long to respond to our request.");
```

Comment:
Shouldn't this be error in place of warn? We are throwing service exception so it should be logged as error, i guess. Same for ConnectionPoolTimeoutException.